renderers/games/teams.py

```diff
@@ -78,11 +85,38 @@ def __render_team_text(canvas, layout, colors, team, homeaway, full_team_names,
     graphics.DrawText(canvas, font["font"], coords["x"], coords["y"], text_color_graphic, team_text)
 
 
-def __render_team_score(canvas, layout, colors, runs, homeaway, default_colors):
+def __render_score_component(canvas, layout, colors, homeaway, default_colors, coords, component_val, width):
+    # return actual calculated x coord.
     text_color = colors.get("text", default_colors["text"])
     text_color_graphic = graphics.Color(text_color["r"], text_color["g"], text_color["b"])
-    coords = layout.coords("teams.runs.{}".format(homeaway))
-    font = layout.font("teams.runs.{}".format(homeaway))
-    team_runs = str(runs)
-    team_runs_x = coords["x"] - (len(team_runs) * font["size"]["width"])
-    graphics.DrawText(canvas, font["font"], team_runs_x, coords["y"], text_color_graphic, team_runs)
+    font = layout.font(f"teams.runs.{homeaway}")  # just use the runs font
+    component_val = str(component_val)
+    std_spacing = True
+    for i in range(width):
+        score_x = coords["x"] - font["size"]["width"] + 1
+        coords["x"] = score_x
+        if len(component_val) > i:
+            c = component_val[len(component_val) - i - 1]
+            graphics.DrawText(canvas, font["font"], score_x, coords["y"], text_color_graphic, c)
```

Comment:
Can you explain this loop? Why is the `DrawText` call inside an if?

Reply:
Also, is the return value of this function different than the score_spacing value? Can we not just re-use that?

Reply:
In retrospect my logic here might be more complex than it needs to be.  Also I didn't really test with scores in the 20's, and I see that the spacing is different between 1's and 2's.  I would like to work out different spacing for 10's and 20's.  I think I will take some more time and refactor this a bit.

Reply:
Is the font really a different width, or does it just have different values?

The nice thing is for testing you can just have game.away_hits() return 22 or something

Reply:
The font is fixed width.  All characters are 3 pixels wide with the exception of the "1" which is drawn in the leftmost two pixels.  So if you're rendering a 10, it looks like there's more space between the 1 and the 0.  I'm fixing this by shifting the 1 over (but only if it's in the 10's column).  Whew!

Reply:
I think this change isn't a good idea. It complicates the code and looks odd if the font used isnt 4x6 (for example, try the 64x64 layout or 128x64). It's not idea for the 4x6 case but I think it's better to not special-case the teens.